src/openrct2/platform/Posix.cpp

```diff
@@ -130,11 +130,7 @@ bool platform_directory_exists(const utf8* path)
     struct stat dirinfo;
     int32_t result = stat(buffer, &dirinfo);
     log_verbose("checking dir %s, result = %d, is_dir = %d", buffer, result, S_ISDIR(dirinfo.st_mode));
-    if ((result != 0) || !S_ISDIR(dirinfo.st_mode))
-    {
-        return false;
-    }
-    return true;
+    return !((result != 0) || !S_ISDIR(dirinfo.st_mode));
```

Comment:
Old notation is easier to read.

Reply:
`return result == 0) && S_ISDIR(dirinfo.st_mode);`?

Reply:
```suggestion
    return result == 0 && S_ISDIR(dirinfo.st_mode);
```

https://en.wikipedia.org/wiki/De_Morgan%27s_laws